### PR TITLE
Show notifications in tabs

### PR DESCRIPTION
**Pull request by kennyt**
px

you won't be able to reproduce this on your own, but that's what it looks like. i'd be open to comments on the implementation. sidebar badge still needs help with css

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @mlunoe

**Comment by daiyiwhale** [comment]
This looks rad, nice!

File: src/js/mixins/TabsMixin.js [hunk 1]
```diff
@@ -93,16 +94,27 @@ const TabsMixin = {
    * @return {Component} React component to render
    */
   tabs_getRoutedItem(customClasses, tab) {
+    let notificationCount = NotificationStore.getNotificationCount(tab);
```
Comment (rcorral) [inline src/js/mixins/TabsMixin.js]: Is `tab` too generic? Meaning, could we have conflicting tab names?

Comment (kennyt) [inline src/js/mixins/TabsMixin.js]: hmm, good point. most of our tab names are named exactly the same as the route name, but there are a few tabs that aren't routed yet. what do you think we should do?

**Comment by jfurrow** [comment]
@kennyt Let me know when you want to go over the styling of this.

File: src/js/mixins/TabsMixin.js [hunk 2]
```diff
@@ -93,16 +94,27 @@ const TabsMixin = {
...
     let tabLabelClass = classNames(customClasses, {'tab-item-label': true});
-
-    return (
+    let link = (
```
Comment (mlunoe) [inline src/js/mixins/TabsMixin.js]: We can just return this directly, if we put it below the `if`-statement

File: src/js/pages/SystemPage.js [hunk 3]
```diff
@@ -84,13 +85,25 @@ class SystemPage extends mixin(TabsMixin) {
   }
 
   getRoutedItem(tab) {
-    return (
+    let notificationCount = NotificationStore.getNotificationCount(tab);
+    let link = (
```
Comment (mlunoe) [inline src/js/pages/SystemPage.js]: Same thing here

**Comment by philipnrmn** [comment]
Hey Kenny, nice work :)

One small CSS issue - in page header tabs, when a notification is added, characters with descenders get cut off:



I was able to fix this with

```css
.badge-container .tab-item-label { line-height: 1em; }
```

**Comment by jfurrow** [comment]
Kenny and I chatted and are going to tackle the styling in another PR, so I'm merging this.

**Pull request merged**